app/components/button.jsx

```diff
@@ -0,0 +1,187 @@
+import React, {useState, useRef} from 'react';
+import { createUseStyles }  from 'react-jss';
+import cx from 'classnames'
+
+/**
+ * Button component(without stretch goal version) that is styled using react-jss.
+ * It supports various states like hover, active, and disabled, 
+ * and can be configured with custom styles, titles, and callbacks.
+ *
+ * @param {Object} props - The props for the Button component.
+ */
+
+function Button(props) {
+
+    const {
+        className = "", // may or may not be passed. Should be applied to the outer most tag, after local classNames
+        style = {}, // may or may not be passed, Should be applied to the outer most tag
+        onDone = () => {}, // a function that is called when the button is clicked.  - if it exists
+        title = "", // text to display on hover
+        disabled = false, 
+        disableOnClick = false, // if true, the button gets disabled after click and stays disabled - prevents resubmission
+        children,
+        ...otherProps
+    } = props;
+
+    const [isDisabled, setIsDisabled] = useState(disabled);
+    const [parentIsHovered, setParentIsHovered] = useState(false);
+
+    const classes = buttonStyles();
+    const combinedClassName = cx(classes.buttonBase, classes[className], className, { 'hover': parentIsHovered });
```

Comment:
after you fixup the other buttons below, you can take out the classes[className].